project/MimaExcludes.scala

```diff
@@ -36,6 +36,12 @@ object MimaExcludes {
 
   // Exclude rules for 2.3.x
   lazy val v23excludes = v22excludes ++ Seq(
+    // SPARK-22372: Make cluster submission use SparkApplication.
+    ProblemFilters.exclude[DirectMissingMethodProblem]("org.apache.spark.deploy.SparkHadoopUtil.getSecretKeyFromUserCredentials"),
```

Comment:
This is only going into spark 2.3 ?

Reply:
The change is 2.3 only.

I've always questioned why `SparkHadoopUtil` is public in the first place. I'm not that worried about removing the functionality in this method because Spark apps don't really need to depend on it; it's easy to call the Hadoop API directly, and the API did nothing before except in YARN mode.